### DISCUSSION TITLE
Problem: can not rename variable in vue file.

### DISCUSSION THREAD

**Discussion by iamxiaojianzheng** [LazyVim/LazyVim#3641] [Q&A]
I tried to rename the variable, but it always failed quickly.

But in my repeated attempts, I found that rename works fine without volar enabled.

![image](https://github.com/LazyVim/LazyVim/assets/29248764/c54a8a86-9c45-44e6-9186-f0a5478b652d)

**Comment by dpetka2001** [comment]
It seems you're using `inc-rename` to do the renaming. Does it work without `inc-rename`?

**Reply by iamxiaojianzheng** [reply]
> It seems you're using `inc-rename` to do the renaming. Does it work without `inc-rename`?您似乎正在使用 `inc-rename` 重命名。没有它能工作 `inc-rename` 吗？

Yes, I just use `<leader>cr`.

**Reply by iamxiaojianzheng** [reply]
I seem to have found the cause of this problem, which should be caused by the volar lsp. 

When I close it, inc-rename is normal.

**Reply by dpetka2001** [reply]
This is something specific to the `inc-rename` plugin then. It's just an Extra that you enabled, because it's not enabled in default LazyVim installation. Maybe you should raise an issue at `inc-rename` plugin repo because this can't be fixed here you know. Also the issue that you opened should maybe closed if it's related to this discussion?

**Reply by iamxiaojianzheng** [reply]
After some tests I have found the cause, but I am not sure if it is the root cause.

As mentioned above, this is a Volar-related exception. 

So I went to volar's GitHub repo and took a look at README.

I found the key to hybridMode, and then I tried to configure it into vue.lua. 

The inc-rename function can be used normally.

**Reply by dpetka2001** [reply]
Cool I only started with Lua and Python 5-6 months ago myself and am not familiar with other languages. Glad you were able to find your way through this :smile:

**Reply by iamxiaojianzheng** [reply]
> Cool I only started with Lua and Python 5-6 months ago myself and am not familiar with other languages. Glad you were able to find your way through this 😄

I should learn from you, but I'm not familiar with lua at the moment.
